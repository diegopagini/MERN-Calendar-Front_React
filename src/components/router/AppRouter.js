/** @format */

import React, { useEffect } from 'react';
import { BrowserRouter as Router, Switch, Redirect } from 'react-router-dom';
import { LoginScreen } from '../auth/LoginScreen';
import { CalendarScreen } from '../calendar/CalendarScreen';
import { useDispatch, useSelector } from 'react-redux';
import { startChecking } from '../../actions/auth';
import { PublicRoute } from './PublicRoute';
import { PrivateRoute } from './PrivateRoute';

export const AppRouter = () => {
	const dispatch = useDispatch();
	const { checking, uid } = useSelector((state) => state.auth);

	useEffect(() => {
		dispatch(startChecking());
	}, [dispatch]);

	if (checking) {
		return <h5>Espere...</h5>;
	}

	return (
		<Router>
			<div>
				<Switch>
					<PublicRoute
						exact
						path='/login'
						component={LoginScreen}
						isAuthenticated={!!uid}
					/>
					<PrivateRoute
						exact
						path='/'
						component={CalendarScreen}
						isAuthenticated={!!uid}
					/>
					<Redirect to='/' />
				</Switch>
			</div>
		</Router>
	);
};
