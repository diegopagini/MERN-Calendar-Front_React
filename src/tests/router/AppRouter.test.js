/** @format */

import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';

import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import '@testing-library/jest-dom';
import { AppRouter } from '../../router/AppRouter';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

// store.dispatch = jest.fn();

describe('Pruebas en <AppRouter />', () => {
	test('debe de mostrar el espere...', () => {
		const initState = {
			auth: {
				checking: true,
			},
		};

		const store = mockStore(initState);

		const wrapper = mount(
			<Provider store={store}>
				<AppRouter />
			</Provider>
		);

		// expect( wrapper ).toMatchSnapshot();
		expect(wrapper.find('h5').exists()).toBe(true);
	});

	test('debe de mostrar la ruta pÃºblica', () => {
		const initState = {
			auth: {
				checking: false,
				uid: null,
			},
		};

		const store = mockStore(initState);

		const wrapper = mount(
			<Provider store={store}>
				<AppRouter />
			</Provider>
		);

		expect(wrapper).toMatchSnapshot();
		expect(wrapper.find('.login-container').exists()).toBe(true);
	});

	test('debe de mostrar la ruta privada', () => {
		const initState = {
			calendar: {
				events: [],
			},
			ui: {
				modalOpen: false,
			},
			auth: {
				checking: false,
				uid: '123',
				name: 'Juan Carlos',
			},
		};

		const store = mockStore(initState);

		const wrapper = mount(
			<Provider store={store}>
				<AppRouter />
			</Provider>
		);

		expect(wrapper).toMatchSnapshot();
		expect(wrapper.find('.calendar-screen').exists()).toBe(true);
	});
});
