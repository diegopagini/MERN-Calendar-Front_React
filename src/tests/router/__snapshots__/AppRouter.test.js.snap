// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AppRouter /> debe de mostrar la ruta privada 1`] = `
<div>
  <div
    className="calendar-screen"
  >
    <div
      className="navbar navbar-dark bg-dark mb-4"
    >
      <span
        className="navbar-brand"
      >
        Juan Carlos
      </span>
      <button
        className="btn btn-outline-danger"
        onClick={[Function]}
      >
        <i
          className="fas fa-sign-out-alt"
        />
        <span>
           Salir
        </span>
      </button>
    </div>
    <div
      className="rbc-calendar"
    >
      <div
        className="rbc-toolbar"
      >
        <span
          className="rbc-btn-group"
        >
          <button
            onClick={[Function]}
            type="button"
          >
            Hoy
          </button>
          <button
            onClick={[Function]}
            type="button"
          >
            &lt;
          </button>
          <button
            onClick={[Function]}
            type="button"
          >
            &gt;
          </button>
        </span>
        <span
          className="rbc-toolbar-label"
        >
          junio 2020
        </span>
        <span
          className="rbc-btn-group"
        >
          <button
            className="rbc-active"
            key="month"
            onClick={[Function]}
            type="button"
          >
            Mes
          </button>
          <button
            className=""
            key="week"
            onClick={[Function]}
            type="button"
          >
            Semana
          </button>
          <button
            className=""
            key="day"
            onClick={[Function]}
            type="button"
          >
            Día
          </button>
          <button
            className=""
            key="agenda"
            onClick={[Function]}
            type="button"
          >
            Agenda
          </button>
        </span>
      </div>
      <div
        className="rbc-month-view"
      >
        <div
          className="rbc-row rbc-month-header"
        >
          <div
            className="rbc-header"
            key="header_0"
          >
            <span>
              lun.
            </span>
          </div>
          <div
            className="rbc-header"
            key="header_1"
          >
            <span>
              mar.
            </span>
          </div>
          <div
            className="rbc-header"
            key="header_2"
          >
            <span>
              mié.
            </span>
          </div>
          <div
            className="rbc-header"
            key="header_3"
          >
            <span>
              jue.
            </span>
          </div>
          <div
            className="rbc-header"
            key="header_4"
          >
            <span>
              vie.
            </span>
          </div>
          <div
            className="rbc-header"
            key="header_5"
          >
            <span>
              sáb.
            </span>
          </div>
          <div
            className="rbc-header"
            key="header_6"
          >
            <span>
              dom.
            </span>
          </div>
        </div>
        <div
          className="rbc-month-row"
        >
          <div
            className="rbc-row-bg"
          >
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
          </div>
          <div
            className="rbc-row-content"
          >
            <div
              className="rbc-row "
            >
              <div
                className="rbc-date-cell"
                key="header_0"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  01
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_1"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  02
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_2"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  03
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_3"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  04
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_4"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  05
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_5"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  06
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_6"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  07
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="rbc-month-row"
        >
          <div
            className="rbc-row-bg"
          >
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
          </div>
          <div
            className="rbc-row-content"
          >
            <div
              className="rbc-row "
            >
              <div
                className="rbc-date-cell"
                key="header_0"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  08
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_1"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  09
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_2"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  10
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_3"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  11
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_4"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  12
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_5"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  13
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_6"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  14
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="rbc-month-row"
        >
          <div
            className="rbc-row-bg"
          >
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg rbc-today"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
          </div>
          <div
            className="rbc-row-content"
          >
            <div
              className="rbc-row "
            >
              <div
                className="rbc-date-cell"
                key="header_0"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  15
                </a>
              </div>
              <div
                className="rbc-date-cell rbc-now rbc-current"
                key="header_1"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  16
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_2"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  17
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_3"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  18
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_4"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  19
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_5"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  20
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_6"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  21
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="rbc-month-row"
        >
          <div
            className="rbc-row-bg"
          >
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
          </div>
          <div
            className="rbc-row-content"
          >
            <div
              className="rbc-row "
            >
              <div
                className="rbc-date-cell"
                key="header_0"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  22
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_1"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  23
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_2"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  24
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_3"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  25
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_4"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  26
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_5"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  27
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_6"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  28
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="rbc-month-row"
        >
          <div
            className="rbc-row-bg"
          >
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg"
            />
            <div
              className="rbc-day-bg rbc-off-range-bg"
            />
            <div
              className="rbc-day-bg rbc-off-range-bg"
            />
            <div
              className="rbc-day-bg rbc-off-range-bg"
            />
            <div
              className="rbc-day-bg rbc-off-range-bg"
            />
            <div
              className="rbc-day-bg rbc-off-range-bg"
            />
          </div>
          <div
            className="rbc-row-content"
          >
            <div
              className="rbc-row "
            >
              <div
                className="rbc-date-cell"
                key="header_0"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  29
                </a>
              </div>
              <div
                className="rbc-date-cell"
                key="header_1"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  30
                </a>
              </div>
              <div
                className="rbc-date-cell rbc-off-range"
                key="header_2"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  01
                </a>
              </div>
              <div
                className="rbc-date-cell rbc-off-range"
                key="header_3"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  02
                </a>
              </div>
              <div
                className="rbc-date-cell rbc-off-range"
                key="header_4"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  03
                </a>
              </div>
              <div
                className="rbc-date-cell rbc-off-range"
                key="header_5"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  04
                </a>
              </div>
              <div
                className="rbc-date-cell rbc-off-range"
                key="header_6"
              >
                <a
                  href="#"
                  onClick={[Function]}
                >
                  05
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <button
      className="btn btn-primary fab"
      onClick={[Function]}
    >
      <i
        className="fas fa-plus"
      />
    </button>
    <Portal
      containerInfo={
        <div
          class="ReactModalPortal"
        />
      }
    />
  </div>
</div>
`;

exports[`Pruebas en <AppRouter /> debe de mostrar la ruta pública 1`] = `
<div>
  <div
    className="container login-container"
  >
    <div
      className="row"
    >
      <div
        className="col-md-6 login-form-1"
      >
        <h3>
          Ingreso
        </h3>
        <form
          onSubmit={[Function]}
        >
          <div
            className="form-group"
          >
            <input
              className="form-control"
              name="lEmail"
              onChange={[Function]}
              placeholder="Correo"
              type="text"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <input
              className="form-control"
              name="lPassword"
              onChange={[Function]}
              placeholder="Contraseña"
              type="password"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <input
              className="btnSubmit"
              type="submit"
              value="Login"
            />
          </div>
        </form>
      </div>
      <div
        className="col-md-6 login-form-2"
      >
        <h3>
          Registro
        </h3>
        <form
          onSubmit={[Function]}
        >
          <div
            className="form-group"
          >
            <input
              className="form-control"
              name="rName"
              onChange={[Function]}
              placeholder="Nombre"
              type="text"
              value="Nando"
            />
          </div>
          <div
            className="form-group"
          >
            <input
              className="form-control"
              name="rEmail"
              onChange={[Function]}
              placeholder="Correo"
              type="email"
              value="nando@gmail.com"
            />
          </div>
          <div
            className="form-group"
          >
            <input
              className="form-control"
              name="rPassword1"
              onChange={[Function]}
              placeholder="Contraseña"
              type="password"
              value="123456"
            />
          </div>
          <div
            className="form-group"
          >
            <input
              className="form-control"
              name="rPassword2"
              onChange={[Function]}
              placeholder="Repita la contraseña"
              type="password"
              value="123456"
            />
          </div>
          <div
            className="form-group"
          >
            <input
              className="btnSubmit"
              type="submit"
              value="Crear cuenta"
            />
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
